maps-compose/src/main/java/com/google/maps/android/compose/Polyline.kt

```diff
@@ -69,12 +70,117 @@ public fun Polyline(
+        color = color,
+        endCap = endCap,
+        geodesic = geodesic,
+        jointType = jointType,
+        pattern = pattern,
+        startCap = startCap,
+        tag = tag,
+        visible = visible,
+        width = width,
+        zIndex = zIndex,
+        onClick = onClick,
+    )
+}
+
+/**
+ * A composable for a polyline on the map.
+ *
+ * @param points the points comprising the polyline
+ * @param spans style spans for the polyline
+ * @param clickable boolean indicating if the polyline is clickable or not
+ * @param endCap a cap at the end vertex of the polyline
+ * @param geodesic specifies whether to draw the polyline as a geodesic
+ * @param jointType the joint type for all vertices of the polyline except the start and end
+ * vertices
+ * @param pattern the pattern for the polyline
+ * @param startCap the cap at the start vertex of the polyline
+ * @param visible the visibility of the polyline
+ * @param width the width of the polyline in screen pixels
+ * @param zIndex the z-index of the polyline
+ * @param onClick a lambda invoked when the polyline is clicked
+ */
+@Composable
+@GoogleMapComposable
+public fun Polyline(
+    points: List<LatLng>,
+    spans: List<StyleSpan>,
+    clickable: Boolean = false,
+    endCap: Cap = ButtCap(),
+    geodesic: Boolean = false,
+    jointType: Int = JointType.DEFAULT,
+    pattern: List<PatternItem>? = null,
+    startCap: Cap = ButtCap(),
+    tag: Any? = null,
+    visible: Boolean = true,
+    width: Float = 10f,
+    zIndex: Float = 0f,
+    onClick: (Polyline) -> Unit = {},
+) {
+    PolylineImpl(
+        points = points,
+        spans = spans,
+        clickable = clickable,
+        endCap = endCap,
+        geodesic = geodesic,
+        jointType = jointType,
+        pattern = pattern,
+        startCap = startCap,
+        tag = tag,
+        visible = visible,
+        width = width,
+        zIndex = zIndex,
+        onClick = onClick,
+    )
+}
+
+/**
+ * Internal implementation for an advanced polyline on a Google map.
+ *
+ * @param points the points comprising the polyline
+ * @param spans style spans for the polyline
+ * @param clickable boolean indicating if the polyline is clickable or not
+ * @param color the color of the polyline
+ * @param endCap a cap at the end vertex of the polyline
+ * @param geodesic specifies whether to draw the polyline as a geodesic
+ * @param jointType the joint type for all vertices of the polyline except the start and end
+ * vertices
+ * @param pattern the pattern for the polyline
+ * @param startCap the cap at the start vertex of the polyline
+ * @param visible the visibility of the polyline
+ * @param width the width of the polyline in screen pixels
+ * @param zIndex the z-index of the polyline
+ * @param onClick a lambda invoked when the polyline is clicked
+ */
+@Composable
+@GoogleMapComposable
+private fun PolylineImpl(
+    points: List<LatLng>,
+    spans: List<StyleSpan> = emptyList(),
```

Comment:
Looks like `PolylineImpl()` could be eliminated entirely in favor of having a single `Polyline()` method with a default parameter. It's a bit of a breaking change, but personally I'd favor it over having so much repetitive overload verbosity, just like with the `Marker()` overloads nightmare. I'm not a maintainer, though.

Reply:
Then there will be confusion with the color parameter - it makes no sense to specify it when we specify Spans for segments